Add mapping for Intech TEK2

Greg-Orca opened the pull request:
TEK2 config files

https://mixxx.zulipchat.com/#narrow/stream/113295-controller-mapping/topic/Intech.20TEK2

fwcd committed:
clang-format: Indent Objective-C blocks with 4 spaces

Commit:
Add track move commands to controls library. Sending key event to library widget.

Co-authored-by: ronso0 <ronso0@mixxx.org>

ronso0 committed:
WOverview: use parented_ptr, QStringLiteral

ronso0 committed:
allow changing the waveform overview type without reloading the skin

ronso0 committed:
2.5 changelog structuring

daschuer committed:
Update changelog for 2.6-alpha

Co-authored-by: ronso0 <ronso0@mixxx.org>

daschuer committed:
Merge pull request #12216 from ronso0/menu-icon-position-fix

Skins / Qt6: menu icon position fix

Commit:
limit to a minimum fontpointsize of 10, add a bit more margin to avoid artifacts visible at 1:1 pixel ratio

Commit:
adjusted factor

Commit:
use std::lround instead of std::lroundf

daschuer committed:
Merge pull request #13311 from m0dB/digits-render-tweaks

Improved readability and scaling of digits and avoid some artefacts

ronso0 committed:
Tooltips: add `bpmlock`

ronso0 committed:
LateNight: deactivate beatgrid edit controls if BPM is locked

daschuer committed:
Merge pull request #13320 from ronso0/bpmlock-latenight

LateNight: deactivate beatgrid edit controls if BPM is locked

daschuer committed:
Merge pull request #13302 from ronso0/pref-opti-apply

Preferences: apply only in slotApply()

acolombier committed:
Add missing const qualifier and add checks on RB retrieve

acolombier committed:
Use QStringLiteral on ScopedTimer

ywwg committed:
Merge pull request #11407 from acolombier/feat/controller-screen-render

feat: controller screen renderer

daschuer committed:
Merge pull request #13313 from ronso0/2.5-changelog

2.5 changelog structuring

daschuer committed:
CHANGELOG: Sort Remaning Feature entries to Interface section

ronso0 committed:
Deere: rework/stabilize key/BPM edit row

daschuer committed:
Add latest changes to CHANGELOG.md

Co-authored-by: ronso0 <ronso0@mixxx.org>

ronso0 committed:
Deere: add BPM lock toggle to beatrid edit row

ronso0 committed:
Deere: disable beatgrid edit controls if BPM is locked

ronso0 committed:
LateNight: simplify template for button with optional cover

ronso0 committed:
(fix) LateNight: correct ObjectName variable

daschuer committed:
Merge pull request #13323 from ronso0/deere-keybpm-controls

Deere: deactivate beatgrid edit controls if BPM is locked + LateNight fix

acolombier committed:
Replace the waitcond and mutex with sema on RBTask

acolombier committed:
Add better comments and various nits

ronso0 committed:
Tango: deactivate beatgrid edit controls if BPM is locked

ronso0 committed:
(fix) Effects: show newly added effects, read/write HiddenEffects

daschuer committed:
Add missing includes

daschuer committed:
Merge pull request #13325 from ronso0/tango-bpmlock

Tango: deactivate beatgrid edit controls if BPM is locked

daschuer committed:
Restore QT5 compatibility for the sake of quick regression checks

daschuer committed:
Fix warning in StemInfoImporter

daschuer committed:
Fix searching for untranslated ConfigKey in input mapping table

acolombier committed:
Fix incorrect checkbox name and reword warning message

fwcd committed:
ControllerRenderingEngine: Fix GL includes

Instead of include <GL/gl.h>, which is unavailable on macOS and not used
elsewhere in the codebase either, we include the Qt OpenGL headers.

fwcd committed:
ControllerScriptEngineLegacy: Disable execution policy with libc++

fwcd committed:
CI: Reenable QML in macOS builds

fwcd committed:
ControllerScriptEngineLegacy: Remove execution policy

As per PR feedback

fwcd committed:
ControllerScriptEngineLegacy: Use reinterpret_cast for image buffer

ronso0 committed:
(opti/cleanup) make use of WLibrary::getCurrentTrackTableView() ...

and move WLibraryTableView::moveSelection to WTracktableView (simplifies LibraryControl etc.)

fwcd committed:
ControllerScriptEngineLegacy: Add short comment on `bit_cast`

fwcd committed:
ControllerScriptEngineLegacy: Update comment on `bit_cast`

fwcd committed:
CI: Disable QML CI on macOS x64 for now

https://github.com/mixxxdj/mixxx/pull/13334#issuecomment-2152990404

Swiftb0y committed:
Merge pull request #13303 from ronso0/autodj-xfader-recenter

add AutoDJ xfader recenter option (default off)

Swiftb0y committed:
Merge pull request #13334 from fwcd/fix-macos-qml-build

Fix QML build regressions on macOS

ronso0 committed:
Search: add 'type' filter

daschuer committed:
Hide Control 0xFF in case of one byte messages

JoergAtGithub committed:
Improve playlist parser test coverage

JoergAtGithub committed:
Fix MSVC warnings in parsercsv.cpp

JoergAtGithub committed:
Merge pull request #13335 from ronso0/library-getCurrentTrackTableView

(opti/cleanup) make use of WLibrary::getCurrentTrackTableView()

JoergAtGithub committed:
Merge pull request #13338 from ronso0/search-by-filetype

Search: add 'type' filter

JoergAtGithub committed:
Added code to allow / path separator on Windows too

daschuer committed:
Merge pull request #13143 from acolombier/feat/multithreaded-rubberband

Multithreaded rubberband

daschuer committed:
Add a missing inner null check

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

Swiftb0y committed:
Merge pull request #13341 from daschuer/midi_input_table

Midi input table fixes

ronso0 committed:
Merge pull request #13317 from daschuer/changelog_26

Update changelog for 2.6-alpha

daschuer committed:
Update Translation template. Found 3104 source texts (176 new and 2928 already existing)

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

Swiftb0y committed:
Merge pull request #13347 from daschuer/qt5_fix

2.5: Qt5 build fix

daschuer committed:
Install missing qml modules

acolombier committed:
Add further asserts, comments and reword misleading symbols

Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>
Signed-off-by: Antoine C <mixxx@acolombier.dev>

acolombier committed:
Refactor linear scaler to use SampleUtil

Signed-off-by: Antoine C <mixxx@acolombier.dev>

acolombier committed:
use SampleUtil::clear to reset RB buffer

acolombier committed:
Merge remote-tracking branch 'upstream/main' into feat/add-engine-support-for-stem

Signed-off-by: Antoine C <mixxx@acolombier.dev>

JoergAtGithub committed:
Made code conditional for Windows

acolombier committed:
Add logic for multi threaded RB

daschuer committed:
Merge pull request #13344 from JoergAtGithub/fix_playlistparser_warning

Fix playlist parser warning with MSVC from VS2022

acolombier committed:
Add missing SampleUtil::copy

acolombier committed:
Rewording RB multi threading to clarify the stereo split into mono

acolombier committed:
chore: update the donate button label

Signed-off-by: Antoine C <mixxx@acolombier.dev>

daschuer committed:
Merge pull request #13322 from daschuer/changelog_25

Changelog for 2.5-beta

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Merge remote-tracking branch 'upstream/2.5' into main

daschuer committed:
Bump CMAKE_PROJECT_VERSION and LICENSE to 2.5-beta

daschuer committed:
Merge pull request #13261 from fonsargo/compressor_effect_fixes

Compressor effect fixes

daschuer committed:
Merge pull request #13356 from daschuer/2.5-beta

2.5-beta Release candidate

JoergAtGithub committed:
Execute expensive setStleSheet only, if textcolor of BPM checkbox(padlock) changed

JoergAtGithub committed:
localDateTimeFromUtc: Make argument a const reference and initialize QDateTime at construction

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

ywwg committed:
Merge pull request #13353 from acolombier/chore/update-donate-button

chore: update the donate button label

JoergAtGithub committed:
Added TODOs

JoergAtGithub committed:
Merge remote-tracking branch 'upstream/2.5'

daschuer committed:
Merge pull request #13358 from JoergAtGithub/Optimize_BpmDelegate

Optimize BPMDelegate::paintItem

daschuer committed:
Merge pull request #13359 from JoergAtGithub/Optimize_localDateTimeFromUtc

localDateTimeFromUtc: Make argument a const reference and initialize QDateTime at construction

ronso0 committed:
WOverview: move redundant drawing code to drawNextPixmapPart()

JoergAtGithub committed:
Pre-allocate memory in basetrackcache to avoid multiple reallocations

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

ronso0 committed:
WOverview: use QStringLiteral for ScopedTimer id

JoergAtGithub committed:
WPixmapStore: Change getPixmapNoCache to std::unique_ptr and further optimizations

ronso0 committed:
use enum class for waveform overview type

daschuer committed:
Update Translation template. Found 3104 source text(s) (0 new and 3104 already existing)

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

Swiftb0y committed:
Merge pull request #13368 from JoergAtGithub/Optimize_basetrackcache

Pre-allocate memory in basetrackcache to avoid multiple reallocations

daschuer committed:
Merge pull request #13273 from ronso0/overview-type-integration

allow changing the waveform overview type without reloading the skin

daschuer committed:
Merge remote-tracking branch 'upstream/2.5' into main

daschuer committed:
Install missing build dependencies for Ubuntu Oracular

The packages libqt6labsqmlmodels6, libqt6quickcontrols2impl6 and libqt6quicklayouts6 have been integrated in the corrosponding qml6-module packages.

daschuer committed:
Update Translation template. Found 3226 source text(s) (38 new and 3078 already existing)

fwcd committed:
CMakeLists: Work around Qt shader bug with Xcode

fwcd committed:
Waveforms: Disable textured waveforms when using OpenGL ES

fwcd committed:
ControllerRenderingEngine: Patch out unavailable APIs when using GL ES

fwcd committed:
AudioUnitManager: Disable unavailable in-process instantiation on iOS

daschuer committed:
Merge pull request #13382 from fwcd/controllerrenderingengine-gles

ControllerRenderingEngine: Patch out unavailable APIs when using GL ES

ronso0 committed:
WOverview: use PollingControlProxy, remove ControlFlag::NoAssertIfMissing

fwcd committed:
WaveformRendererSignalBase: Disable HighDetail on OpenGL ES

daschuer committed:
Merge pull request #13379 from fwcd/fix-qt-shader-bug-xcode

CMakeLists: Work around Qt shader bug with Xcode

daschuer committed:
Merge pull request #13383 from fwcd/disable-audiounit-in-process-ios

AudioUnitManager: Disable unavailable in-process instantiation on iOS

dependabot[bot] committed:
Bump actions/checkout from 4.1.6 to 4.1.7

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.6 to 4.1.7.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4.1.6...v4.1.7)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

daschuer committed:
Merge pull request #13386 from mixxxdj/dependabot/github_actions/actions/checkout-4.1.7

Bump actions/checkout from 4.1.6 to 4.1.7

danferns committed:
add KeyDelegate

danferns committed:
add Default Key Color Palette

danferns committed:
use Key Color Palette to change Key Column background

ronso0 committed:
Merge pull request #13098 from nlw0/nic/track-move-controls

Add track move commands to controls library

daschuer committed:
Merge pull request #13381 from fwcd/disable-textured-waveforms-gles

Waveforms: Disable textured waveforms when using OpenGL ES

danferns committed:
implement the design for Key Column colors

danferns committed:
style: increase right padding for key color rectangle

acolombier committed:
Add loop vectorisation comments and reword misleading sentences

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.5'

acolombier committed:
Refactor linearCrossfadeStemBuffersOut to add vectorization

acolombier committed:
Merge remote-tracking branch 'upstream/main' into feat/add-engine-support-for-stem

acolombier committed:
Remove misleading tooltip message

ChrisTallon committed:
Draw minute markers on horizontal waveform overview widget

Also add a new setting in preferences/waveforms to enable turning
minute markers on and off.

daschuer committed:
Merge pull request #13070 from acolombier/feat/add-engine-support-for-stem

Add support for stem in the engine

alhadebe committed:
add controller settings for the Numark Scratch

acolombier committed:
feat: add support for STEM file in analyzers

This will allow analyser to make the most of the multiple channels
to analyze specific features of the track. It will also generate a multi
channel waveform

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

ronso0 committed:
(fix) Library: show Properties action in Missing and Hidden

danferns committed:
refactor: cleanup #includes and move KeyDelegate constructor into header

danferns committed:
fix: use selected text color in KeyDelegate

danferns committed:
move keyToColor translation into KeyUtils

also check bounds for Open Key number inside it.

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>
Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>

daschuer committed:
Avoid plain new in WColorPickerAction ctor and make sure m_pColorPicker has always a parent.

daschuer committed:
Move the parent assertion back to the parented_ptr ctor to not longer allow the weak situation where an object is not memory managed. This reverts df7706406b78eb244d29014f65ac963afa444f0d partially.

daschuer committed:
Add parent during construction in DlgLibraryExport

daschuer committed:
Fix memory leak in createFindOnWebSubmenus()

daschuer committed:
Fix missing parent in LegacyControllerSettingsLayoutContainer

daschuer committed:
Fix find on web menu tree

daschuer committed:
parented_ptr: Add a compile time check for constructiong with a parent

danferns committed:
style: remove left padding of key color bars

This will allow for narrow column setups.

daschuer committed:
Add missing parent

Swiftb0y committed:
Merge pull request #13411 from daschuer/improve_parented_ptr

Improve use of parented_ptr

JoergAtGithub committed:
auto* to indicate ptr

Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>

JoergAtGithub committed:
Use QStringLiteral instead of QString

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

JoergAtGithub committed:
Don't use .get()

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

JoergAtGithub committed:
Use std::unique_ptr instead of QScopedPointer and use std::move to transfer it with ownership

JoergAtGithub committed:
Replaced Qt ptr types by std ones and get rid of unnecessary new

danferns committed:
refactor: pass KEY_ID to KeyDelegate, instead of Key String

Co-authored-by: ronso0 <ronso0@mixxx.org>

acolombier committed:
AnalyzerSilence: use multi channel signal without stereo mixing

acolombier committed:
AnalyzerKey: add comment about further multi channel source support

acolombier committed:
Remove the outdated kAnalysisChannels constant

acolombier committed:
Remove scaleSignal for stem

JoergAtGithub committed:
Merge pull request #13370 from ronso0/overview-type-enum

use enum class for waveform overview type

JoergAtGithub committed:
Merge pull request #13326 from ronso0/effects-visible-list

(fix) Effects: show newly added effects, read/write HiddenEffects

JoergAtGithub committed:
Use using instead of typedef

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

danferns committed:
add checkbox to toggle key colors

Add a `KeyColorsEnabled` setting under the `[Config]` group. And render a checkbox for it in the Color preferences.

JoergAtGithub committed:
Pass references to underlying type instead of managed pointers

acolombier committed:
Handle Dolby channel count usecase in the analyzer

JoergAtGithub committed:
Made Paintable::DrawMode an enum class

JoergAtGithub committed:
Removed unused setSVG and hash functionality from pixmapsource.
Instead use the getPath explicit as in key.
Improved cache key string concatenation

acolombier committed:
fix: invalid slip render marker

Swiftb0y committed:
Merge pull request #13369 from JoergAtGithub/Optimize_WPixmapStore

WPixmapStore: Change getPixmapNoCache to std::unique_ptr and further optimizations

Swiftb0y committed:
Merge pull request #13422 from acolombier/fix/slip-render-marker

fix: invalid slip render marker

JoergAtGithub committed:
Changed enum values to TitleCase

JoergAtGithub committed:
Changed hash keys to structs

christophehenry committed:
Don't return in JogWheelBasic on deck absent in option

JoergAtGithub committed:
Use autogenerated default equal operator

Greg-Orca committed:
Add mapping for Intech TEK2

TEK2 config files

https://mixxx.zulipchat.com/#narrow/stream/113295-controller-mapping/topic/Intech.20TEK2

Greg-Orca commented:
based on previous PR https://github.com/mixxxdj/mixxx/pull/13405

ronso0 committed:
WOverview: fix right minute markers in vertical overviews

Swiftb0y committed:
Merge pull request #13423 from JoergAtGithub/cleanupPixmapSource

Removed unused setSVG and hash functionality from pixmapsource

Swiftb0y committed:
Merge pull request #13401 from ChrisTallon/main-minutemarkers

Add minute markers on horizontal waveform overview

JoergAtGithub committed:
Implemented fast map lookup for backward compatible DrawMode strings

JoergAtGithub committed:
Merge remote-tracking branch 'upstream/main' into PaintableDrawmodeEnumClass

JoergAtGithub committed:
Use early return style

danferns committed:
feat: toggle key colors from preferences

Swiftb0y committed:
Merge pull request #13424 from JoergAtGithub/PaintableDrawmodeEnumClass

Made Paintable::DrawMode an enum class

danferns committed:
fix: checkbox ui not in sync with key colors setting

danferns committed:
refactor: remove redundant canConvert assertion

danferns committed:
mention source of Mixxx Key Color Palette colors

danferns committed:
fix indentation and set colspan in Color Preferences UI

danferns committed:
remove unnecessary slotUpdate() call

JoergAtGithub committed:
Removed 2 unused methods
Changed code to return early style
Implemented missing SVG path for Paintable::toImage()
CReplaced hardcoded transparency color by Qt::transparent
Use std::move instead reset/release

Swiftb0y commented:
Thank you for your Pull request. Please understand that in order for us to be able to merge this, your contribution need to follow [our style guide](https://github.com/mixxxdj/mixxx/wiki/Coding%20Guidelines#mixxx-style-guide). Most of the required changes can be made automatically by use of `pre-commit`. Please [set up pre-commit](https://pre-commit.com/#installation) and make sure to manually apply the required changes using `pre-commit run --from-ref 2.4 --to-ref HEAD`. Thank you very much.

daschuer committed:
Merge pull request #13426 from ronso0/track-prop-in-hidden-and-missing

(fix) Library: show Properties action in Missing and Hidden

JoergAtGithub committed:
Use Format_ARGB32_Premultiplied, which is generaly faster than Format_ARGB32 for images where the content itself is not modified.

JoergAtGithub committed:
Added comments why we need to fill the QImage data before we can draw on it.

JoergAtGithub committed:
Use QPixmap::fromImage to directly construct the QPixmap from the QImage

daschuer committed:
Merge pull request #13435 from JoergAtGithub/painable3

Paintable cleanup

ronso0 committed:
Library: fix font reset in multiline comment editor

acolombier committed:
Add user preferences to only enabled stem optimisation on demand

acolombier committed:
Make supported stem constant global

ronso0 committed:
(fix) Sound preferences: don't set m_settingsModified in update slots

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Update Translation template. Found XXXX source text(s) (0 new and 3104 already existing)

ronso0 committed:
Pref Sound: use PollingControlProxy and parented_ptr

dependabot[bot] committed:
Bump actions/upload-artifact from 4.3.3 to 4.3.4

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4.3.3 to 4.3.4.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4.3.3...v4.3.4)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Merge pull request #13452 from mixxxdj/dependabot/github_actions/actions/upload-artifact-4.3.4

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

daschuer committed:
Merge remote-tracking branch 'upstream/2.5' into main

daschuer committed:
Update Translation template. Found 3118 source text(s) (8 new and 3110 already existing)

christophehenry committed:
Deprecate lodash.mixxx.js and script.deepMerge for removal in a future release

Swiftb0y committed:
fix: add missing ControlObject forward-decl

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-6/. Compile QM files out of TS files that are used by the localized app

daschuer committed:
Add missing include

daschuer committed:
Add an optional seed to the hash functions to keep full QT compliance and remove the std namespace block.

danferns committed:
refactor: set keyColor, keyText as const

Co-authored-by: ronso0 <ronso0@mixxx.org>

danferns committed:
refactor: remove KeyUtils scope used in keyToColor

Co-authored-by: ronso0 <ronso0@mixxx.org>

danferns committed:
refactor: update key colors checkbox name

also add checkbox to <tabstops>

Co-authored-by: ronso0 <ronso0@mixxx.org>

danferns committed:
clear keyText when key is invalid

Swiftb0y committed:
Merge pull request #13458 from daschuer/hash_clean_up

hash clean up

danferns committed:
refactor: draw key text/color only if key is valid

Swiftb0y committed:
Merge pull request #13404 from NotYourAverageAl/NumarkScratchSettings

feat(controllers): add controller settings for the Numark Scratch

danferns committed:
remove #define KEY_COLORS_ENABLED

m0dB committed:
Merge pull request #13448 from ronso0/lib-comment-editor-font-fix

Library: fix font reset in multiline comment editor

daschuer committed:
Merge pull request #13390 from danferns/static_key_colors

feat: static color coding for key column

acolombier committed:
Hide the stem analyzer setting

acolombier committed:
Various feedback from code review

Eve00000 committed:
Add files via upload

JoergAtGithub committed:
Merge pull request #13106 from acolombier/feat/add-analyser-support-for-stem

Add analyser support for stem

daschuer committed:
Merge pull request #13468 from Eve00000/20240713-NL-Dutch-gui-translations-update

20240713 NL Dutch gui translations updated (+/- 300 items)

daschuer committed:
Update Translation template. Found 3126 source text(s) (8 new and 3118 already existing)

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-6/. Compile QM files out of TS files that are used by the localized app

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

daschuer committed:
Merge remote-tracking branch 'upstream/2.5' into main

acolombier committed:
fix: support for new WaveformData struct in shaders

danferns committed:
fix: Key text is elided from left, should be right

Swiftb0y committed:
Merge pull request #13475 from danferns/key_elided_text

fix: Key text is elided from left, should be right

ywwg committed:
Merge pull request #13474 from acolombier/fix/waveform-shader-support-for-new-struct

fix: support for new WaveformData struct in shaders

christophehenry committed:
Fix deprecation notice on VSCode

fwcd committed:
SoundManagerIOS: Remove unsupported/redundant options

ronso0 committed:
Overview waveform: draw minute markers on top of played overlay

daschuer committed:
Merge pull request #13450 from ronso0/soundpref-apply-keylockdualthread

(fix) Sound preferences: don't set m_settingsModified in update slots

ywwg committed:
fix high details waveforms wrapping around after visual index 65K

Fixes https://github.com/mixxxdj/mixxx/issues/13472

JoergAtGithub committed:
Merge pull request #13489 from ronso0/overview-minute-ontop

Overview waveform: draw minute markers on top of played overlay

m0dB committed:
Merge pull request #13487 from fwcd/soundmanagerios-remove-redundant-options

SoundManagerIOS: Remove unsupported/redundant options

JoergAtGithub committed:
Merge pull request #13460 from christophehenry/depreciate-lodash

Deprecate `lodash.mixxx.js`, and `script.deepMerge`

dependabot[bot] committed:
Bump azure/trusted-signing-action from 0.3.20 to 0.4.0

Bumps [azure/trusted-signing-action](https://github.com/azure/trusted-signing-action) from 0.3.20 to 0.4.0.
- [Release notes](https://github.com/azure/trusted-signing-action/releases)
- [Commits](https://github.com/azure/trusted-signing-action/compare/v0.3.20...v0.4.0)

---
updated-dependencies:
- dependency-name: azure/trusted-signing-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

fwcd committed:
ControllerRenderingEngine: Disable BGRA when targeting Wasm

JoergAtGithub committed:
Merge pull request #13503 from fwcd/objc-block-indent

clang-format: Indent Objective-C blocks with 4 spaces

JoergAtGithub committed:
Merge pull request #13502 from fwcd/disable-bgra-wasm

ControllerRenderingEngine: Disable BGRA when targeting Wasm

Swiftb0y committed:
Merge pull request #13500 from mixxxdj/dependabot/github_actions/azure/trusted-signing-action-0.4.0

Bump azure/trusted-signing-action from 0.3.20 to 0.4.0

Holzhaus committed:
fix(basetracktablemodel): Fix `-Wimplicit-fallthrough` warning on GCC 14.1.1

Fixes the following warning:

    src/library/basetracktablemodel.cpp: In member function ‘QVariant BaseTrackTableModel::roleValue(const QModelIndex&, QVariant&&, int) const’:
    src/library/basetracktablemodel.cpp:1031:9: error: this statement may fall through [-Werror=implicit-fallthrough=]
     1031 |         switch (field) {
          |         ^~~~~~
    src/library/basetracktablemodel.cpp:1042:5: note: here
     1042 |     default:
          |     ^~~~~~~
    cc1plus: all warnings being treated as errors

This seems to be a false-positive, but this commit should fix the issue
without any changes to the behavior.

Greg-Orca committed:
formatting

Greg-Orca committed:
Merge branch 'mixxxdj:main' into intech-tek2

Greg-Orca committed:
Merge branch '2.4' into intech-tek2

Swiftb0y committed:
Merge pull request #13505 from Holzhaus/gcc-14.1.1-fixes-main

fix(basetracktablemodel): Fix `-Wimplicit-fallthrough` warning on GCC 14.1.1

Swiftb0y committed:
Merge pull request #13425 from christophehenry/dont-return-on-no-deck-JogWheelBasic

Don't return in JogWheelBasic on deck absent in option

JoergAtGithub committed:
Merge pull request #13491 from ywwg/ywwg/fix-high-details

fix high details waveforms wrapping around after visual index 65K

Greg-Orca committed:
Merge branch 'mixxxdj:main' into intech-tek2
